### ISSUE TITLE
Flaky Tekton tests

### ISSUE THREAD

**Issue by roytman** [fybrik/fybrik#1032]
The CI Tekton tests occasionally fails. 
For example:
https://github.com/fybrik/fybrik/runs/4517803129?check_suite_focus=true
This is the same rerunning test, but with different results
https://github.com/fybrik/fybrik/runs/4518405872?check_suite_focus=true

**Comment by ngoracke** [comment]
Unit tests appear to be hanging, or at least taking long enough that we hit the 30 minute timeout for the run.  From the output below, they were running 12 minutes before we gave up:

```
+ kubectl get taskrun,pvc,po
NAME                                                                              SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
taskrun.tekton.dev/build-and-deploy-run-build-datacatalog-blsfq                   True        Succeeded   30m         18m
taskrun.tekton.dev/build-and-deploy-run-build-image-datacatalog-f5sd6             True        Succeeded   18m         17m
taskrun.tekton.dev/build-and-deploy-run-build-image-katalog-swjgw                 True        Succeeded   23m         21m
taskrun.tekton.dev/build-and-deploy-run-build-image-manager-7gqn4                 True        Succeeded   13m         12m
taskrun.tekton.dev/build-and-deploy-run-build-image-opa-h2n5g                     True        Succeeded   21m         20m
taskrun.tekton.dev/build-and-deploy-run-build-image-serverpolicymanager-vgqdq     True        Succeeded   18m         17m
taskrun.tekton.dev/build-and-deploy-run-build-katalog-49gh2                       True        Succeeded   30m         23m
taskrun.tekton.dev/build-and-deploy-run-build-manager-g4fww                       True        Succeeded   30m         13m
taskrun.tekton.dev/build-and-deploy-run-build-opa-xtlhz                           True        Succeeded   30m         21m
taskrun.tekton.dev/build-and-deploy-run-build-servicepolicymanager-wdcwd          True        Succeeded   30m         18m
taskrun.tekton.dev/build-and-deploy-run-fetch-repository-w8fls                    True        Succeeded   30m         30m
taskrun.tekton.dev/build-and-deploy-run-run-integration-tests-full-deploy-qh8h2   True        Succeeded   12m         65s
taskrun.tekton.dev/build-and-deploy-run-run-unit-tests-full-deploy-74xb4          Unknown     Running     12m         
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-certmgr-from-repo-vvn72      True        Succeeded   30m         29m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-crd-from-source-vcck5        True        Succeeded   30m         29m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-full-cd9ts                   True        Succeeded   12m         12m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-vault-from-repo-s29nj        True        Succeeded   30m         27m
```

From the unit test logs, we see the start of the tests...

```
2021/12/14 09:50:34 Start Mock for Catalog Connector at :50085
2021/12/14 09:50:39 Start Mock for OPA Server at :8383
```

and the logs end here...

```
=== RUN   TestSerde
Running Suite: Serde Suite
==========================
Random Seed: 1639475639
Will run 4 of 4 specs

••••
Ran 4 of 4 Specs in 0.001 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestSerde (0.00s)
PASS
ok  	fybrik.io/fybrik/pkg/serde	0.025s
?   	fybrik.io/fybrik/pkg/slices	[no test files]
?   	fybrik.io/fybrik/pkg/storage	[no test files]
```

**Comment by ngoracke** [comment]
In the passed run, it does look like we finish just under the 30 minute timeout

```
+ kubectl get taskrun,pvc,po
NAME                                                                              SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
taskrun.tekton.dev/build-and-deploy-run-build-datacatalog-v4xxd                   True        Succeeded   29m         17m
taskrun.tekton.dev/build-and-deploy-run-build-image-datacatalog-bqlhv             True        Succeeded   17m         16m
taskrun.tekton.dev/build-and-deploy-run-build-image-katalog-wnd65                 True        Succeeded   21m         19m
taskrun.tekton.dev/build-and-deploy-run-build-image-manager-2gbt5                 True        Succeeded   13m         12m
taskrun.tekton.dev/build-and-deploy-run-build-image-opa-xqknp                     True        Succeeded   17m         16m
taskrun.tekton.dev/build-and-deploy-run-build-image-serverpolicymanager-gs9q5     True        Succeeded   18m         17m
taskrun.tekton.dev/build-and-deploy-run-build-katalog-vc9jk                       True        Succeeded   29m         21m
taskrun.tekton.dev/build-and-deploy-run-build-manager-w8nk7                       True        Succeeded   29m         13m
taskrun.tekton.dev/build-and-deploy-run-build-opa-j9lnx                           True        Succeeded   29m         17m
taskrun.tekton.dev/build-and-deploy-run-build-servicepolicymanager-bncc7          True        Succeeded   29m         18m
taskrun.tekton.dev/build-and-deploy-run-fetch-repository-c4pbg                    True        Succeeded   29m         29m
taskrun.tekton.dev/build-and-deploy-run-run-integration-tests-full-deploy-p6xgx   True        Succeeded   12m         91s
taskrun.tekton.dev/build-and-deploy-run-run-unit-tests-full-deploy-pgd69          Unknown     Running     12m         
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-certmgr-from-repo-5dh42      True        Succeeded   29m         27m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-crd-from-source-tt82q        True        Succeeded   29m         28m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-full-4kxdb                   True        Succeeded   12m         12m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-vault-from-repo-c6tz4        True        Succeeded   29m         27m

NAME                                  STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
persistentvolumeclaim/artifacts-pvc   Bound    pvc-bc155bab-9e40-4160-a067-6738a4a49e25   500Mi      RWX            nfs-client     31m
persistentvolumeclaim/charts-pvc      Bound    pvc-a6784d86-4804-4558-b2a5-d78c8a76cf77   500Mi      RWX            nfs-client     31m
persistentvolumeclaim/source-pvc      Bound    pvc-84cdf7bd-a0b4-47cf-9d85-c6052ef947a0   1000Mi     RWX            nfs-client     31m

NAME                                                                  READY   STATUS      RESTARTS   AGE
pod/affinity-assistant-c11e0e815a-0                                   1/1     Running     0          29m
pod/affinity-assistant-c8949013e2-0                                   1/1     Running     0          29m
pod/apiserversource-tekton-task-wat6e7469af32b849eec08e9f496c6jmpg6   1/1     Running     0          31m
pod/build-and-deploy-run-build-datacatalog-v4xxd-pod-6v4n9            0/3     Completed   0          29m
pod/build-and-deploy-run-build-image-datacatalog-bqlhv-pod-kvdcq      0/8     Completed   0          17m
pod/build-and-deploy-run-build-image-katalog-wnd65-pod-npf66          0/8     Completed   0          21m
pod/build-and-deploy-run-build-image-manager-2gbt5-pod-pr9fh          0/8     Completed   0          13m
pod/build-and-deploy-run-build-image-opa-xqknp-pod-qdmb9              0/8     Completed   0          17m
pod/build-and-deploy-run-build-image-serverpolicymanager-gs9q-dwxwc   0/8     Completed   0          18m
pod/build-and-deploy-run-build-katalog-vc9jk-pod-dztl2                0/3     Completed   0          29m
pod/build-and-deploy-run-build-manager-w8nk7-pod-wv66z                0/3     Completed   0          29m
pod/build-and-deploy-run-build-opa-j9lnx-pod-zhhf2                    0/3     Completed   0          29m
pod/build-and-deploy-run-build-servicepolicymanager-bncc7-pod-4zvz4   0/3     Completed   0          29m
pod/build-and-deploy-run-fetch-repository-c4pbg-pod-qfl5m             0/1     Completed   0          29m
pod/build-and-deploy-run-run-integration-tests-full-deploy-p6-rf8vc   0/3     Completed   0          12m
pod/build-and-deploy-run-run-unit-tests-full-deploy-pgd69-pod-gwflb   3/3     Running     0          12m
pod/build-and-deploy-run-upgrade-helm-certmgr-from-repo-5dh42-dbzfm   0/1     Completed   0          29m
pod/build-and-deploy-run-upgrade-helm-crd-from-source-tt82q-p-5hc54   0/1     Completed   0          29m
pod/build-and-deploy-run-upgrade-helm-full-4kxdb-pod-lbn98            0/1     Completed   0          12m
pod/build-and-deploy-run-upgrade-helm-vault-from-repo-c6tz4-p-q8bcl   0/3     Completed   0          29m
pod/katalog-connector-56d5bf5d95-vb7w7                                1/1     Running     0          12m
pod/manager-75669586dc-trhqr                                          2/2     Running     0          12m
pod/opa-6d8f95cd7c-g7rts                                              2/2     Running     0          12m
pod/opa-connector-5f976c569f-hbjkb                                    1/1     Running     0          12m
pod/vault-0                                                           1/1     Running     0          27m
pod/workspace-0                                                       1/1     Running     0          31m
+ kubectl get pipelinerun --no-headers
build-and-deploy-run   Unknown   Running   29m   
+ kubectl get pipelinerun --no-headers
+ grep -e Failed -e Completed
+ '[' 1 -ne 0 ']'
+ sleep 30
+ (( TRIES++  ))
+ (( TRIES<=60 ))
+ /tmp/tmp.f4IePR19sT/streams_csv_check_script.sh
+ kubectl get taskrun,pvc,po
NAME                                                                              SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
taskrun.tekton.dev/build-and-deploy-run-build-datacatalog-v4xxd                   True        Succeeded   29m         18m
taskrun.tekton.dev/build-and-deploy-run-build-image-datacatalog-bqlhv             True        Succeeded   18m         17m
taskrun.tekton.dev/build-and-deploy-run-build-image-katalog-wnd65                 True        Succeeded   21m         19m
taskrun.tekton.dev/build-and-deploy-run-build-image-manager-2gbt5                 True        Succeeded   13m         13m
taskrun.tekton.dev/build-and-deploy-run-build-image-opa-xqknp                     True        Succeeded   18m         17m
taskrun.tekton.dev/build-and-deploy-run-build-image-serverpolicymanager-gs9q5     True        Succeeded   19m         18m
taskrun.tekton.dev/build-and-deploy-run-build-katalog-vc9jk                       True        Succeeded   29m         21m
taskrun.tekton.dev/build-and-deploy-run-build-manager-w8nk7                       True        Succeeded   29m         13m
taskrun.tekton.dev/build-and-deploy-run-build-opa-j9lnx                           True        Succeeded   29m         18m
taskrun.tekton.dev/build-and-deploy-run-build-servicepolicymanager-bncc7          True        Succeeded   29m         19m
taskrun.tekton.dev/build-and-deploy-run-fetch-repository-c4pbg                    True        Succeeded   30m         29m
taskrun.tekton.dev/build-and-deploy-run-finally-remove-creds-9g7rd                True        Succeeded   24s         5s
taskrun.tekton.dev/build-and-deploy-run-run-integration-tests-full-deploy-p6xgx   True        Succeeded   12m         2m2s
taskrun.tekton.dev/build-and-deploy-run-run-unit-tests-full-deploy-pgd69          True        Succeeded   12m         24s
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-certmgr-from-repo-5dh42      True        Succeeded   30m         28m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-crd-from-source-tt82q        True        Succeeded   29m         28m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-full-4kxdb                   True        Succeeded   13m         12m
taskrun.tekton.dev/build-and-deploy-run-upgrade-helm-vault-from-repo-c6tz4        True        Succeeded   30m         28m

NAME                                  STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
persistentvolumeclaim/artifacts-pvc   Bound    pvc-bc155bab-9e40-4160-a067-6738a4a49e25   500Mi      RWX            nfs-client     32m
persistentvolumeclaim/charts-pvc      Bound    pvc-a6784d86-4804-4558-b2a5-d78c8a76cf77   500Mi      RWX            nfs-client     32m
persistentvolumeclaim/source-pvc      Bound    pvc-84cdf7bd-a0b4-47cf-9d85-c6052ef947a0   1000Mi     RWX            nfs-client     32m

NAME                                                                  READY   STATUS        RESTARTS   AGE
pod/affinity-assistant-c11e0e815a-0                                   0/1     Terminating   0          30m
pod/affinity-assistant-c8949013e2-0                                   0/1     Terminating   0          30m
pod/apiserversource-tekton-task-wat6e7469af32b849eec08e9f496c6jmpg6   1/1     Running       0          31m
pod/build-and-deploy-run-build-datacatalog-v4xxd-pod-6v4n9            0/3     Completed     0          29m
pod/build-and-deploy-run-build-image-datacatalog-bqlhv-pod-kvdcq      0/8     Completed     0          18m
pod/build-and-deploy-run-build-image-katalog-wnd65-pod-npf66          0/8     Completed     0          21m
pod/build-and-deploy-run-build-image-manager-2gbt5-pod-pr9fh          0/8     Completed     0          13m
pod/build-and-deploy-run-build-image-opa-xqknp-pod-qdmb9              0/8     Completed     0          18m
pod/build-and-deploy-run-build-image-serverpolicymanager-gs9q-dwxwc   0/8     Completed     0          19m
pod/build-and-deploy-run-build-katalog-vc9jk-pod-dztl2                0/3     Completed     0          29m
pod/build-and-deploy-run-build-manager-w8nk7-pod-wv66z                0/3     Completed     0          29m
pod/build-and-deploy-run-build-opa-j9lnx-pod-zhhf2                    0/3     Completed     0          29m
pod/build-and-deploy-run-build-servicepolicymanager-bncc7-pod-4zvz4   0/3     Completed     0          29m
pod/build-and-deploy-run-fetch-repository-c4pbg-pod-qfl5m             0/1     Completed     0          30m
pod/build-and-deploy-run-finally-remove-creds-9g7rd-pod-5bsg8         0/1     Completed     0          24s
pod/build-and-deploy-run-run-integration-tests-full-deploy-p6-rf8vc   0/3     Completed     0          12m
pod/build-and-deploy-run-run-unit-tests-full-deploy-pgd69-pod-gwflb   0/3     Completed     0          12m
pod/build-and-deploy-run-upgrade-helm-certmgr-from-repo-5dh42-dbzfm   0/1     Completed     0          30m
pod/build-and-deploy-run-upgrade-helm-crd-from-source-tt82q-p-5hc54   0/1     Completed     0          29m
pod/build-and-deploy-run-upgrade-helm-full-4kxdb-pod-lbn98            0/1     Completed     0          13m
pod/build-and-deploy-run-upgrade-helm-vault-from-repo-c6tz4-p-q8bcl   0/3     Completed     0          30m
pod/katalog-connector-56d5bf5d95-vb7w7                                1/1     Running       0          13m
pod/manager-75669586dc-trhqr                                          2/2     Running       0          13m
pod/opa-6d8f95cd7c-g7rts                                              2/2     Running       0          13m
pod/opa-connector-5f976c569f-hbjkb                                    1/1     Running       0          13m
pod/vault-0                                                           1/1     Running       0          28m
pod/workspace-0                                                       1/1     Running       0          32m
+ kubectl get pipelinerun --no-headers
build-and-deploy-run   True   Completed   30m   4s
+ kubectl get pipelinerun --no-headers
+ grep -e Failed -e Completed
build-and-deploy-run   True   Completed   30m   4s
+ '[' 0 -ne 0 ']'
```

**Comment by roytman** [comment]
Hi @ngoracke, thanks for the comments. 
Is it possible to change the tests to reduce their running time?

**Comment by roytman** [comment]
here is another example when at the first attempt texton failed after 4m , and successfully finished at the second attempt.
https://github.com/fybrik/fybrik/actions/runs/1582362103/attempts/1
https://github.com/fybrik/fybrik/actions/runs/1582362103

**Comment by shlomitk1** [comment]
Unit tests alone do not take much time - the "Unit tests" workflow runs ~3min. 'make test' runs less than 2 min on my machine. Integration tests run longer because  most of the time is spent on building images and activating pods, so it takes ~8 min. 
@ngoracke Is it possible to print out the logs of this pod, or is there a single command I can run to debug this?

**Comment by shlomitk1** [comment]
> here is another example when at the first attempt texton failed after 4m , and successfully finished at the second attempt. https://github.com/fybrik/fybrik/actions/runs/1582362103/attempts/1 https://github.com/fybrik/fybrik/actions/runs/1582362103

This seems to be the reason:
`Error from server (InternalError): error when creating "/home/runner/work/fybrik/fybrik/pipeline/../pipeline/eventlistener/eventlistener.yaml": Internal error occurred: failed calling webhook "webhook.triggers.tekton.dev": Post "https://tekton-triggers-webhook.tekton-pipelines.svc:443/defaulting?timeout=10s": dial tcp 10.96.120.210:443: connect: connection refused`

**Comment by ngoracke** [comment]
I should have time this weekend to look at this further, thanks for the patience!

**Comment by ngoracke** [comment]
https://github.com/fybrik/fybrik/pull/1089

Spent some time executing unit tests & integration tests in serial.  On my laptop or a "real" kube cluster, running in parallel seems to be no problem.  There appears to be significant throttling taking place on the github actions infra/kind cluster, however.

This change will not increase the total runtime of the test, but unit tests seem to be consistently passing in 6 minutes (from pod creation to completion).

This will not address the connection refused message posted above.  I'll look at this separately -

**Comment by roytman** [comment]
Hi @ngoracke , can you try to run Tekton tests as a separate GitHub action, it might help.

**Comment by roee88** [comment]
See https://github.com/fybrik/fybrik/runs/4685138547?check_suite_focus=true as a new example

**Comment by ngoracke** [comment]
https://github.com/fybrik/fybrik/pull/1115/files

This should address the other failure in the tekton action (attempting to apply task definitions when webhook controller appears to be up, but is not yet accepting requests).

**Comment by roytman** [comment]
Hi, the tests continue to fail.
see https://github.com/fybrik/fybrik/runs/4711696892?check_suite_focus=true

**Comment by shlomitk1** [comment]
I see this in the logs: `/bin/sh: docker: command not found`

**Comment by roytman** [comment]
> Hi, the tests continue to fail. see https://github.com/fybrik/fybrik/runs/4711696892?check_suite_focus=true

indeed it is another issue, thanks @shlomitk1

**Comment by ngoracke** [comment]
https://github.com/fybrik/fybrik/pull/1125

This was a new issue introduced by https://github.com/fybrik/fybrik/pull/1068

Although this was worked around by a recent commit from @roytman , the change above will allow calls to docker to work from a tekton task

**Comment by roytman** [comment]
Hi @ngoracke , you can see another Tekton pipeline failure [here](https://github.com/roytman/fybrik/runs/4804436398?check_suite_focus=true).

**Comment by ngoracke** [comment]
This is new, from the build datacatalog task.  Would you expect datacatalog to be doing something with docker?

```
+ make source-build
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o datacatalog datacatalogstub.go
go: downloading github.com/gin-gonic/gin v1.7.7
go: downloading github.com/rs/zerolog v1.26.0
go: downloading k8s.io/api v0.22.4
go: downloading k8s.io/apimachinery v0.22.4
go: downloading sigs.k8s.io/controller-runtime v0.9.5
go: downloading emperror.dev/errors v0.7.0
go: downloading google.golang.org/grpc v1.41.0
go: downloading github.com/mohae/deepcopy v0.0.0-20170929034955-c48cc78d4826
go: downloading github.com/gin-contrib/sse v0.1.0
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/xeipuuv/gojsonschema v1.2.0
go: downloading k8s.io/apiextensions-apiserver v0.22.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/google/gofuzz v1.2.0
go: downloading k8s.io/klog v1.0.0
go: downloading sigs.k8s.io/structured-merge-diff v1.0.1-0.20191108220359-b1b620dd3f06
go: downloading k8s.io/klog/v2 v2.9.0
go: downloading gomodules.xyz/jsonpatch/v2 v2.2.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.2
go: downloading k8s.io/client-go v0.22.4
go: downloading github.com/pkg/errors v0.9.1
go: downloading go.uber.org/multierr v1.6.0
go: downloading golang.org/x/oauth2 v0.0.0-20210514164344-f6687ab2804c
go: downloading google.golang.org/protobuf v1.27.1
go: downloading golang.org/x/net v0.0.0-20210825183410-e898025ed96a
go: downloading github.com/go-playground/validator/v10 v10.4.1
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/ugorji/go v1.1.7
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/ugorji/go/codec v1.1.7
go: downloading golang.org/x/sys v0.0.0-20211020174200-9d6173849985
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
go: downloading k8s.io/utils v0.0.0-20210819203725-bdf08cb9a70a
go: downloading k8s.io/apiserver v0.22.4
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/google/go-cmp v0.5.6
go: downloading github.com/go-logr/logr v0.4.0
go: downloading github.com/prometheus/client_golang v1.11.0
go: downloading github.com/json-iterator/go v1.1.11
go: downloading github.com/fsnotify/fsnotify v1.5.1
go: downloading go.uber.org/atomic v1.9.0
go: downloading google.golang.org/genproto v0.0.0-20210707164411-8c882eb9abba
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/go-playground/universal-translator v0.18.0
go: downloading github.com/leodido/go-urn v1.2.1
go: downloading golang.org/x/crypto v0.0.0-20210711020723-a769d52b0f97
go: downloading github.com/xeipuuv/gojsonpointer v0.0.0-20190905194746-02993c407bfb
go: downloading k8s.io/kube-openapi v0.0.0-20211109043538-20434351676c
go: downloading github.com/evanphx/json-patch v4.11.0+incompatible
go: downloading golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
go: downloading github.com/imdario/mergo v0.3.12
go: downloading github.com/spf13/pflag v1.0.5
go: downloading golang.org/x/term v0.0.0-20210220032956-6a3ed077a48d
go: downloading k8s.io/component-base v0.22.4
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/common v0.29.0
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading go.opentelemetry.io/otel v0.20.0
go: downloading go.opentelemetry.io/otel/trace v0.20.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/go-playground/locales v0.14.0
go: downloading github.com/go-openapi/jsonreference v0.19.5
go: downloading github.com/go-openapi/swag v0.19.14
go: downloading github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d
go: downloading github.com/mitchellh/mapstructure v1.4.2
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/googleapis/gnostic v0.5.5
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading github.com/google/uuid v1.2.0
go: downloading sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.22
go: downloading github.com/blang/semver v3.5.1+incompatible
go: downloading go.opentelemetry.io/contrib v0.20.0
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.20.0
go: downloading go.opentelemetry.io/otel/exporters/otlp v0.20.0
go: downloading go.opentelemetry.io/otel/sdk v0.20.0
go: downloading golang.org/x/text v0.3.6
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading go.opentelemetry.io/otel/metric v0.20.0
go: downloading go.opentelemetry.io/otel/sdk/export/metric v0.20.0
go: downloading go.opentelemetry.io/otel/sdk/metric v0.20.0
go: downloading go.opentelemetry.io/proto/otlp v0.7.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.16.0
+ echo -n fybrik-system
+ echo -n kind-registry:5000
time="2022-01-13T14:26:08.766421271Z" level=warning msg="could not change group /var/run/docker.sock to docker: group docker not found"
time="2022-01-13T14:26:08.766538573Z" level=warning msg="[!] DON'T BIND ON ANY IP ADDRESS WITHOUT setting --tlsverify IF YOU DON'T KNOW WHAT YOU'RE DOING [!]"
time="2022-01-13T14:26:08.800377950Z" level=info msg="libcontainerd: started new docker-containerd process" pid=22
time="2022-01-13T14:26:08Z" level=info msg="starting containerd" module=containerd revision=773c489c9c1b21a6d78b5c538cd395416ec50f88 version=v1.0.3 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.content.v1.content"..." module=containerd type=io.containerd.content.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.snapshotter.v1.btrfs"..." module=containerd type=io.containerd.snapshotter.v1 
time="2022-01-13T14:26:08Z" level=warning msg="failed to load plugin io.containerd.snapshotter.v1.btrfs" error="path /var/lib/docker/containerd/daemon/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter" module=containerd 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.snapshotter.v1.overlayfs"..." module=containerd type=io.containerd.snapshotter.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.metadata.v1.bolt"..." module=containerd type=io.containerd.metadata.v1 
time="2022-01-13T14:26:08Z" level=warning msg="could not use snapshotter btrfs in metadata plugin" error="path /var/lib/docker/containerd/daemon/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter" module="containerd/io.containerd.metadata.v1.bolt" 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.differ.v1.walking"..." module=containerd type=io.containerd.differ.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.gc.v1.scheduler"..." module=containerd type=io.containerd.gc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.containers"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.content"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.diff"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.events"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.healthcheck"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.images"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.leases"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.namespaces"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.snapshots"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.monitor.v1.cgroups"..." module=containerd type=io.containerd.monitor.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.runtime.v1.linux"..." module=containerd type=io.containerd.runtime.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.tasks"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.version"..." module=containerd type=io.containerd.grpc.v1 
time="2022-01-13T14:26:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.introspection"..." module=containerd type=io.containerd.grpc.v1 
+ set +e
+ kubectl get pipelinerun --no-headers
time="2022-01-13T14:26:08Z" level=info msg=serving... address="/var/run/docker/containerd/docker-containerd-debug.sock" module="containerd/debug" 
time="2022-01-13T14:26:08Z" level=info msg=serving... address="/var/run/docker/containerd/docker-containerd.sock" module="containerd/grpc" 
time="2022-01-13T14:26:08Z" level=info msg="containerd successfully booted in 0.074114s" module=containerd 
Failed to connect to containerd: failed to dial "/var/run/docker/containerd/docker-containerd.sock": dial unix /var/run/docker/containerd/docker-containerd.sock: connect: connection refused
```

**Comment by ngoracke** [comment]
At first glance, it looks like Docker (backing Kind) may have been unresponsive.  Going to keep digging.  Nothing in datacatalog's build should be touching docker itself.

**Comment by ngoracke** [comment]
yeah, so:

When we implemented https://github.com/fybrik/fybrik/pull/1125 we introduced a docker sidecar to the "make" tasks.

In many cases, we do not need that sidecar to be up and running to complete a build.  We would only need it if a call to docker is made (currently, for doc builds - we use buildah for other image builds).

In this failure, the sidecar attempted to start, but the docker socket (from the kind cluster worker node) was not responding quick enough to requests.

Options:

1) Only bring up the sidecar for tasks that require it, rather than just having it handy for any Make task.  It's still possible docker could be too slow to respond when a sidecar is brought up, but this would greatly reduce the likelihood.  The only downside to this I see is that we'd need to be explicit about when tasks in the pipeline require it.  Seems like a small downside.

2) Brute force with more retries.

3) A combination of #1 and #2.

i'll submit a PR with option 1, and we can discuss further whether or not it is acceptable.

**Comment by roytman** [comment]
Thanks @ngoracke , your PR was merged, hopefully, it will resolve the issue.

**Comment by roytman** [comment]
it failed again, https://github.com/fybrik/fybrik/actions/runs/1705104095.
maybe we should give up on the dind support.

**Comment by ngoracke** [comment]
```
+ kubectl apply -f /home/runner/work/fybrik/fybrik/pipeline/../pipeline/eventlistener/eventlistener.yaml
Error from server (InternalError): error when creating "/home/runner/work/fybrik/fybrik/pipeline/../pipeline/eventlistener/eventlistener.yaml": Internal error occurred: failed calling webhook "webhook.triggers.tekton.dev": Post "https://tekton-triggers-webhook.tekton-pipelines.svc:443/defaulting?timeout=10s": dial tcp 10.96.131.37:443: connect: connection refused
```

This is unrelated to the DIND support.  Rather, tekton webhook didn't respond quickly enough to a creation request.  While this is not necessary on a real kube/openshift cluster, we seem to be getting rate limited quite a bit.  What I'd like to do is add a retry for any of the commands in bootstrap that need to be processed by tekton's webhook.

**Comment by ngoracke** [comment]
I'll submit a PR for this in the morning - looks like the 'retry' worked ok at least?

**Comment by roytman** [comment]
Thanks, indeed `retry` usually works

**Comment by ngoracke** [comment]
I'll give this a number of runs through before we merge

https://github.com/fybrik/fybrik/pull/1180

**Comment by roytman** [comment]
thanks

**Comment by roytman** [comment]
we don't observe the issue now. 
Thanks @ngoracke.
